ocp_resources/resource.py

```diff
@@ -107,13 +108,32 @@ def get_client(
             )
         )
     try:
-        # Ref: https://github.com/kubernetes-client/python/blob/v26.1.0/kubernetes/base/config/__init__.py
-        LOGGER.info("Trying to get client via new_client_from_config")
-
         # kubernetes.config.kube_config.load_kube_config sets KUBE_CONFIG_DEFAULT_LOCATION during module import.
         # If `KUBECONFIG` environment variable is set via code, the `KUBE_CONFIG_DEFAULT_LOCATION` will be None since
         # is populated during import which comes before setting the variable in code.
         config_file = config_file or os.environ.get("KUBECONFIG", "~/.kube/config")
+        proxy = kwargs.pop("proxy", None) or os.environ.get("HTTPS_PROXY") or os.environ.get("HTTP_PROXY")
+
+        if proxy:
```

Comment:
here you just need to prepare `client_configuration` and use the call that we already have in line 137

Reply:
now that I think of it, we should get `client_configuration` from the user and use it
This way we can support any configuration

Reply:
.

Reply:
Done.